### ISSUE TITLE
Error when installing Event dispatcher component and Symfony framework

### ISSUE THREAD

**Issue by matheusdallrosa** [symfony/symfony#29153]
**Symfony version(s) affected**: 3.4.14

**Description**  
<!-- A clear and concise description of the problem. -->
I'm trying to install Symfony v3.4.14 and event-dispatcher v4.1.4, but i'm getting the following error:

```bash
    Your requirements could not be resolved to an installable set of packages. 
                                                                                                                                                                                                                                                                                                                                                                                               
    Problem 1                                                                                                                                                                                                                           
      - Installation request for symfony/event-dispatcher v4.1.4 -> satisfiable by symfony/event-dispatcher[v4.1.4].                                                                                                                    
      - don't install symfony/symfony v3.4.14|don't install symfony/event-dispatcher v4.1.4                                                                                                                                             
      - Installation request for symfony/symfony v3.4.14 -> satisfiable by symfony/symfony[v3.4.14]. 
````

**How to reproduce**  
I'm using this list of packages:
```json
    "require": {
        "php": ">=7.2.0",
        "symfony/symfony": "3.4.14",
        "doctrine/orm": "^2.5",
        "doctrine/doctrine-bundle": "^1.6",
        "doctrine/doctrine-cache-bundle": "^1.2",
        "symfony/swiftmailer-bundle": "^2.3",
        "symfony/monolog-bundle": "^2.8",
        "symfony/polyfill-apcu": "^1.0",
        "doctrine/mongodb-odm-bundle": "^3.0",
        "sensio/distribution-bundle": "^5.0",
        "sensio/framework-extra-bundle": "^3.0.2",
        "incenteev/composer-parameter-handler": "^2.0",
        "friendsofsymfony/rest-bundle": "^2.0",
        "jms/serializer-bundle": "^2.4",
        "symfony/assetic-bundle": "^2.8",
        "knplabs/knp-paginator-bundle": "^2.5",
        "gedmo/doctrine-extensions": "^2.4",
        "doctrine/doctrine-migrations-bundle": "^1.2",
        "hashids/hashids": "^2.0",
        "beberlei/DoctrineExtensions": "^1.0",
        "symfony/event-dispatcher": "4.1.4",
        "aws/aws-sdk-php": "^3.54",
        "ramsey/uuid": "^3.7",
        "kreait/firebase-php": "^4.0",
        "alcaeus/mongo-php-adapter": "^1.1",
        "nelmio/cors-bundle": "^1.5"
    }
````

I've removed the composer.lock and then i tried this: 
```bash
export SYMFONY_ENV='prod';
php composer.phar install --verbose --prefer-dist --no-progress --no-dev --optimize-autoloader
````

**Additional context**  
This command is being generated by the php Deployer tool during the default vendor's step.

**Comment by xabbuh** [comment]
The `symfony/symfony` package already includes the EventDispatcher component. You cannot install the component both in version 3.4 and 4.1 (as you will not know whether you get a class from one version or the other at runtime). Composer wrongly allowed this in the past, but this was fixed in 1.7.3. You need to explicitly choose which version of the EventDispatcher component you want to use.

I am closing here as there is nothing we can do about it in the Symfony code.

**Comment by matheusdallrosa** [comment]
> The `symfony/symfony` package already includes the EventDispatcher component. You cannot install the component both in version 3.4 and 4.1 (as you will not know whether you get a class from one version or the other at runtime). Composer wrongly allowed this in the past, but this was fixed in 1.7.3. You need to explicitly choose which version of the EventDispatcher component you want to use.
> 
> I am closing here as there is nothing we can do about it in the Symfony code.

You were right. The php deployer was downloading the newest version of composer which is the 1.7.3. I removed the EventDispatcher component package and then i could do composer install again. Thank you.
